src/overlays/actors/ovl_Boss_Ganondrof/z_boss_ganondrof.c

```diff
@@ -61,71 +148,1395 @@ static ColliderCylinderInit D_80914CFC = {
+    }
+}
+
+void BossGanondrof_SetupDeath(BossGanondrof* this, GlobalContext* globalCtx) {
+    Animation_PlayOnce(&this->skelAnime, &gPhantomGanonAnim_00090C);
+    this->endFrame = Animation_GetLastFrame(&gPhantomGanonAnim_00090C);
+    this->actionFunc = BossGanondrof_Death;
+    Audio_SetBGM(0x100100FF);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_FANTOM_DEAD);
+    this->deathState = DEATH_START;
+    this->actor.flags &= ~1;
+    this->varianceTimer = 0;
+    this->shockTimer = 50;
+}
+
+void BossGanondrof_Death(BossGanondrof* this, GlobalContext* globalCtx) {
+    u8 holdCamera = false;
+    u8 bodyDecayLevel = 0;
+    f32 camX;
+    f32 camZ;
+    f32 pad;
+    Player* player = PLAYER;
+    Camera* camera = Gameplay_GetCamera(globalCtx, 0);
+
+    osSyncPrintf("PYP %f\n", player->actor.floorHeight);
+    SkelAnime_Update(&this->skelAnime);
+    this->deathSfxTimer++;
+    if (((60 < this->deathSfxTimer) && (this->deathSfxTimer < 500)) ||
+        ((501 < this->deathSfxTimer) && (this->deathSfxTimer < 620))) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_LAST - SFX_FLAG);
+    }
+
+    switch (this->deathState) {
+        case DEATH_START:
+            func_80064520(globalCtx, &globalCtx->csCtx);
+            func_8002DF54(globalCtx, &this->actor, 1);
+            this->deathCamera = Gameplay_CreateSubCamera(globalCtx);
+            Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+            osSyncPrintf("7\n");
+            Gameplay_ChangeCameraStatus(globalCtx, this->deathCamera, 7);
+            osSyncPrintf("8\n");
+            this->deathState = DEATH_THROE;
+            player->actor.speedXZ = 0.0f;
+            this->timers[0] = 50;
+            this->cameraEye = camera->eye;
+            this->cameraAt = camera->at;
+            this->cameraNextEye.x = this->targetPos.x;
+            this->cameraNextEye.y = 50.0f;
+            this->cameraNextEye.z = (this->targetPos.z + 100.0f) + 50.0f;
+            this->cameraNextAt.x = this->targetPos.x;
+            this->cameraNextAt.y = this->targetPos.y - 10.0f;
+            this->cameraNextAt.z = this->targetPos.z;
+            this->cameraEyeVel.x = fabsf(camera->eye.x - this->cameraNextEye.x);
+            this->cameraEyeVel.y = fabsf(camera->eye.y - this->cameraNextEye.y);
+            this->cameraEyeVel.z = fabsf(camera->eye.z - this->cameraNextEye.z);
+            this->cameraAtVel.x = fabsf(camera->at.x - this->cameraNextAt.x);
+            this->cameraAtVel.y = fabsf(camera->at.y - this->cameraNextAt.y);
+            this->cameraAtVel.z = fabsf(camera->at.z - this->cameraNextAt.z);
+            this->cameraAccel = 0.02f;
+            this->cameraEyeMaxVel.x = this->cameraEyeMaxVel.y = this->cameraEyeMaxVel.z = 0.05f;
+            this->actionState = DEATH_SPASM;
+            this->timers[0] = 150;
+            this->cameraAtMaxVel.x = 0.2f;
+            this->cameraAtMaxVel.y = 0.2f;
+            this->cameraAtMaxVel.z = 0.2f;
+        case DEATH_THROE:
```

Comment:
is throe a typo?

Reply:
Nope,

throe (n) -- intense or violent pain and struggle, especially accompanying birth, death, or great change.

Reply:
lmfao, new word to me.
maybe theres another way to describe it thats a bit less niche. or its just on me for never knowing that word before lol, maybe others can weigh in

Reply:
As you might expect, I think this is a reasonable word to use here. However, it is many, many times more common in the plural ( https://books.google.com/ngrams/graph?content=throe%2Cthroes&year_start=1800&year_end=2019&corpus=26&smoothing=3 ), though, so I wonder if that might be more suitable, especially since it covers several subcases anyway.